### DISCUSSION TITLE
Feature Request - forwarded email alias

### DISCUSSION THREAD

**Discussion by darkknight7777777** [dani-garcia/vaultwarden#2534] [Ideas]
I know this feature was just added in bitwarden official last month, but would be awesome to get integrated into vaultwarden. 

https://bitwarden.com/blog/add-privacy-and-security-using-email-aliases-with-bitwarden/

**Comment by BlackDex** [comment]
This is already there with the latest release v1.25.0.
And we also just updated the web-vault to the latest on the testing tagged images.

**Reply by darkknight7777777** [reply]
thanks, I've spun up a test image and verified this is available. Any chance we can edit the anonaddy instance URL? I host my own AA instance.

**Reply by BlackDex** [reply]
Isn't that as easy as just changing the default hostname??
It's just two input fields, one for the API key, the other the hostname.

**Reply by darkknight7777777** [reply]
The 2nd field appears to be for the domain name you want to generate the username for, not the anonaddy instance URL.  
![vwt](https://user-images.githubusercontent.com/87158641/172051417-d7e9afa8-91df-4539-8cc1-fdae2516c7b6.jpg)

I tried putting my instance URL in there and still got an 'Unknown Anonaddy error occurred' message. I also don't see anything in the log when trace is turned on and this is attempted.

**Reply by BlackDex** [reply]
That's because all is done client side. Nothing is send to the server regarding this. So, is probably not something we can fix for you.

**Reply by darkknight7777777** [reply]
Maybe I'm not following, the API works fine with other products (extensions, android app) because I can change the anonaddy instance on the client. I would think having the same ability in vaultwarden would allow selfhosted anonaddy instances work as well.

**Reply by BlackDex** [reply]
May i ask where you can change that? Since I'm not seeing an option anywhere, it is also not mentioned any where if and how you can change the API Endpoint for Anonaddy. 

I have checked the desktop, extension and web-vault (Non Vaultwarden) and i can find it nowhere.
So, if you would be so kind on telling me where we could find that, maybe that will help.

When looking at the code, these URL's are hardcoded.
The only thing i can think of, now that i see the code is that we are missing some default URL's in the Policy we sent.
That is maybe something we need to fix. But other than that, you can't configure a custom endpoint as far as i can tell, in any client.
But please, let us know where, then maybe we can add this feature.

**Reply by darkknight7777777** [reply]
Sorry I was unclear. I am able to change the anonaddy instance URL in the anonaddy browser extension and android app and verify the anonaddy API works there. 

I do not see this is in any bitwarden clients. That is where I would need it for the functionality to work for a selfhosted anonaddy instance I believe.

**Reply by BlackDex** [reply]
For that you need to go to Bitwarden it self since they maintain the clients, and that would also be better for the whole Bitwarden community.

**Reply by darkknight7777777** [reply]
Understood, thank you

**Reply by kmahyyg** [reply]
> The 2nd field appears to be for the domain name you want to generate the username for, not the anonaddy instance URL. ![vwt](https://user-images.githubusercontent.com/87158641/172051417-d7e9afa8-91df-4539-8cc1-fdae2516c7b6.jpg)
> 
> I tried putting my instance URL in there and still got an 'Unknown Anonaddy error occurred' message. I also don't see anything in the log when trace is turned on and this is attempted.

May I know where can I find this settings? I've already use v1.25.0 now.

**Reply by darkknight7777777** [reply]
https://github.com/dani-garcia/vaultwarden/discussions/2534#discussioncomment-2884350

**Reply by kmahyyg** [reply]
> [#2534 (comment)](https://github.com/dani-garcia/vaultwarden/discussions/2534#discussioncomment-2884350)

Well, it seems external forwarding service only available through client instead of web vault currently.

**Comment by kmahyyg** [comment]
Where can I see the anonaddy settings? I can't find it...

**Reply by leon1995** [reply]
I am using the following versions
```
Server Installed 1.25.0 
Server Latest 1.25.0 
Web Installed 2.28.1 
```
and the option is also missing for me ![missing_option](https://user-images.githubusercontent.com/15685899/173785525-d137c2f0-d46e-4131-a169-edf973ee01ca.png)

**Comment by BlackDex** [comment]
This feature is available in the 2022.5.0 version of the web-vault, which is available in the current `testing` tagged images and not `latest`.

**Reply by leon1995** [reply]
Ah okay, because u said `This is already there with the latest release v1.25.0.`
This kinda confused me.
Do you know, when it will come to `latest`?

**Comment by BlackDex** [comment]
Well, it's a bit confusing, since some clients have it in 1.25 and some in later version. Also not all relays are available in the web-vault for example.

**Reply by leon1995** [reply]
But how is this possible?

What is the `testing` tag? Is this version unstable or why is it `testing`? Can I use `testing` for production until the forwarding feature is included in `latest`?

**Reply by BlackDex** [reply]
I suggest to read this: https://github.com/dani-garcia/vaultwarden/wiki/Which-container-image-to-use
In short:
- `latest` are tagged released versions.
- `testing` are based upon the last commit done on the `main` branch.

We try to keep all working as should with the `testing`/`main` branch of course, but there could be some small bugs or regression in there depending on the changes we have done. I my self use `testing` most of the time, and sometimes even a custom build image based upon an other branch when I'm testing the stability of some new features or large changes.

It's all up to you what you prefer to use.
We do not have a specific release schedule, as that all depends on the changes done if there are any, since this project is totally done by developers during there free time.
